test/catalog_test_util.py

```diff
@@ -230,7 +227,9 @@ def tearDown(self):
             self.dockerclient.remove_image(image['Id'])
 
         # make sure NMS is clean after each test
-        self.mongo.drop_database(self.nms_test_cfg['method-spec-mongo-dbname'])
+        # Drop the database set by the method_spec_mongo_dbname environment variable
+        # in the docker-compose_nms.yml file.
+        self.mongo.drop_database('method_store_repo_db')
```

Comment:
This is dropping a database in the catalog mongo db, not the one started up via the docker compose file. Does this run after every test or only at the end of the test suite? If the latter, I don't think it's needed since run_tests.sh does a `docker compose down`. If it is this file will need to set up a mongo client specifically for the NMS.

Reply:
This is indeed dropping a database in the catalog MongoDB. But why would it be placed under the [NarrativeMethodStore] section?

The tearDown class method is run at the end of each test file, so I'm not 100% sure of its intention, as it could go either way. Dropping a database in the catalog MongoDB makes more sense to me because the catalog MongoDB handle was set up in the init helper method.

If that's the case, I would suggest deleting comments L230-231 and moving method_spec_mongo_dbname under the [catalog-test] section, unless you have a different perspective.

Reply:
> But why would it be placed under the [NarrativeMethodStore] section?

Because is the name of the NMS database, I presume.

> Dropping a database in the catalog MongoDB makes more sense to me because the catalog MongoDB handle was set up in the init helper method.

I'm not understanding the argument here - IIUC, the original code dropped the NMS database for the NMS server. At that point, the NMS and catalog shared a Mongo instance. Now they have separate instances and so dropping the NMS database from the catalog's mongo does nothing, because there's nothing to drop. The NMS database is now on the Mongo instance started by the docker compose file, and so needs to be dropped in that Mongo instance.

Am I missing something here?

Reply:
> make sure NMS is clean after each test

Yeah, this comment confirms that it's actually deleting the database from the NMS.

Reply:
👍

Reply:
:+1: